internal/gatewayapi/translator.go

```diff
@@ -758,6 +760,107 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
 						break // If an invalid filter type has been configured then skip processing any more filters
 					}
 					switch filter.Type {
+					case v1beta1.HTTPRouteFilterURLRewrite:
```

Comment:
can you please move this to a function so its easier to review, thanks !

Reply:
Yep, I will refactor the func.

Reply:
Hi @arkodg, since you have mentioned this more than one time, I think this is really need to resolve now, like https://github.com/envoyproxy/gateway/pull/717#discussion_r1043571291 here too. So I refactored the whole gateway-api translator, lines of translator reduce from 2000 to 150. Codes of gateway-api translator are more organized, easy-to-extend, reading-friendly now.